windows/CodePush.cs

```diff
@@ -0,0 +1,668 @@
+        private readonly string DOWNLOAD_PROGRESS_EVENT_NAME = "CodePushDownloadProgress";
+        private readonly string FAILED_UPDATES_KEY = "CODE_PUSH_FAILED_UPDATES";
+        private static readonly string FILE_BUNDLE_PREFIX = "ms-appdata:///local";
+        private static readonly string PACKAGE_HASH_KEY = "packageHash";
+        private readonly string PENDING_UPDATE_HASH_KEY = "hash";
+        private readonly string PENDING_UPDATE_IS_LOADING_KEY = "isLoading";
+        private readonly string PENDING_UPDATE_KEY = "CODE_PUSH_PENDING_UPDATE";
+
+        // This needs to be kept in sync with https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/devsupport/DevSupportManager.java#L78
+        private readonly string REACT_DEV_BUNDLE_CACHE_FILE_NAME = "ReactNativeDevBundle.js";
+
+        // Helper classes.
+        private CodePushNativeModule codePushNativeModule;
+        private CodePushPackage codePushPackage;
+
+        // Config properties.
+        private string appVersion;
+        private string deploymentKey;
+        private readonly string serverUrl = "https://codepush.azurewebsites.net/";
+
+        private ReactPage mainPage;
+
+        private static CodePushModule currentInstance;
+
+        public CodePushModule(string deploymentKey, ReactPage mainPage)
+        {
+            codePushPackage = new CodePushPackage();
+            // TODO implement telemetryManager 
+            // this.codePushTelemetryManager = new CodePushTelemetryManager(this.applicationContext, CODE_PUSH_PREFERENCES);
+            this.deploymentKey = deploymentKey;
+            this.mainPage = mainPage;
+            appVersion = Package.Current.Id.Version.Major + "." + Package.Current.Id.Version.Minor + "." + Package.Current.Id.Version.Build;
+            InitializeUpdateAfterRestart();
+            if (currentInstance != null)
+            {
+                CodePushUtils.log("More than one CodePush instance has been initialized. Please use the instance method codePush.getBundleUrlInternal() to get the correct bundleURL for a particular instance.");
+            }
+
+            currentInstance = this;
+        }
+
+        private async Task ClearReactDevBundleCache()
+        {
+            StorageFile devBundleCacheFile = null;
+            try
+            {
+                devBundleCacheFile = await ApplicationData.Current.LocalFolder.GetFileAsync(REACT_DEV_BUNDLE_CACHE_FILE_NAME);
+            }
+            catch (FileNotFoundException)
+            {
+            }
+
+            if (devBundleCacheFile != null)
+            {
+                await devBundleCacheFile.DeleteAsync();
+            }
+        }
+
+        private async Task<long> GetBinaryResourcesModifiedTime()
+        {
+            StorageFile assetJSBundleFile = await StorageFile.GetFileFromApplicationUriAsync(new Uri(ASSETS_BUNDLE_PREFIX + assetsBundleFileName));
```

Comment:
could you add `this.` to the instance variables so that its obvious where the value is coming from?

`this.assetsBundleFileName`